### PR TITLE
Added support for /__/metrics

### PR DESCRIPTION
**Pull request by catalin-ilea**
Updated the api to be able to add prometheus metrics. The metrics are
exposed on the /__/metrics endpoint.

### CODE REVIEW

File: op/opstatus.go [hunk 1]
```diff
@@ -14,59 +20,71 @@ func NewStatus(name, description string) *Status {
...
-	return os
+func (s *Status) AddChecker(name string, checkerFunc func(cr *CheckResponse)) *Status {
+	s.checkers = append(s.checkers, checker{name, checkerFunc})
+	return s
+}
+
+// AddMetrics registers prometheus metrics to be exopsed on the /__/metrics endpoint
+// Adding the same metric twice will result in a panic
+func (s *Status) AddMetrics(cs ...prometheus.Collector) *Status {
+	if s.promRegistry == nil {
```
Comment (JamieAP) [inline op/opstatus.go]: Does this mean it is not possible to use the default registry? 

That means (and maybe this isn't a bad thing) we can't do `prometheus.MustRegister` from anywhere in code and have the metrics appear on the endpoint. We need to have the OpHandler in scope.

Comment (mjgarton) [inline op/opstatus.go]: Good point. Agreed it isn't a bad thing, because it forces people to keep things reasonably visible.

Comment (mjgarton) [inline op/opstatus.go]: Out of interest @JamieAP, do you find yourself currently using `prometheus.MustRegister` much ?

Comment (catalin-ilea) [inline op/opstatus.go]: Yes, it means metrics registered with the default registry will not appear on the /__/metrics response. But that's exactly what I had in mind. 
IMO forcing clients to register metrics via AddMetrics() only provides better control over the registered metrics.

Comment (JamieAP) [inline op/opstatus.go]: @mjgarton Exclusively. 

@catalin-ilea I agree it's likely a desirable change in terms of forcing people to write better code. Does complicate upgrade paths slightly for existing projects but if you've weighted that in then LGTM.

**Pull request merged**